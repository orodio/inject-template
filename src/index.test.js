import { inject } from './index'

test('inject', () => {
  const str = inject`
    Hello ${ d => d.user.name } the ${ d => d.user.job }.
    How is your ${ d => d.pet.type } named ${ d => d.pet.name }?
    This is a random string ${ "injection!!!" }.
  `

  const data = {
    user: {
      name: 'bob',
      job: 'builder',
    },
    pet: {
      name: 'Pilchard',
      type: 'cat',
    }
  }

  expect(str(data)).toMatchSnapshot()
})
