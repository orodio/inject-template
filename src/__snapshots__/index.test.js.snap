// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inject 1`] = `
"
    Hello bob the builder.
    How is your cat named Pilchard?
    This is a random string injection!!!.
  "
`;
